Testnet/PrivateYesNoVote/PrivateYesNoVote/PrivateYesNoVote.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Stratis.SmartContracts;
+
+public class PrivateYesNoVote : SmartContract
+{
+    public PrivateYesNoVote(ISmartContractState smartContractState, ulong duration, byte[] addressesBytes) 
+        : base(smartContractState)
+    {
+        VotePeriodEndBlock = checked(Block.Number + duration);
+        Owner = Message.Sender;
+        WhiteListAddressesExecute(addressesBytes);
+    }
+
+    public Address Owner
+    {
+        get => PersistentState.GetAddress(nameof(Owner));
+        private set => PersistentState.SetAddress(nameof(Owner), value); 
+    }
+
+    public ulong VotePeriodEndBlock
+    {
+        get => PersistentState.GetUInt64(nameof(VotePeriodEndBlock));
+        private set => PersistentState.SetUInt64(nameof(VotePeriodEndBlock), value);
+    }
+
+    public uint YesVotes
+    {
+        get => PersistentState.GetUInt32(nameof(YesVotes));
+        private set => PersistentState.SetUInt32(nameof(YesVotes), value);
+    }
+    
+    public uint NoVotes
+    {
+        get => PersistentState.GetUInt32(nameof(NoVotes));
+        private set => PersistentState.SetUInt32(nameof(NoVotes), value);
+    }
+
+    private void SetAuthorization(Address address)
+    {
+        PersistentState.SetBool($"Voter:{address}", true);
+    }
+
+    public bool IsAuthorized(Address address)
+    {
+        return PersistentState.GetBool($"Voter:{address}");
+    }
+
+    public char GetVote(Address address)
+    {
+        return PersistentState.GetChar($"Vote:{address}");
+    }
+
+    public void WhitelistAddresses(byte[] addressBytes)
```

Comment:
Can you add another overload method which takes single address parameter ? It is easier to use this way and most likely  single parameter will be called usually.

Reply:
Overload or singular naming preference?

```
WhitelistAddress(Address address) vs WhitelistAddresses(Address address)
```

I like singular naming rather than overload. What do you thinkg?

Also, should I add Blacklist methods? Incase an address is accidentally whitelisted and needs to be removed by the contract owner?

Reply:
Both the terms `Authorize` and `Whitelist` are being used. Maybe just settle on one - `Authorize`.

This means we rename `WhitelistAddresses` to `AuthorizeAddresses` and add another method `RevokeAddresses`.
Rename:
- `SetAuthorization` to `AuthorizeAddressExecute`. 
- `WhiteListAddressesExecute` to `AuthorizeAddressesExecute`.
Add public method `AuthorizeAddress` that checks sender and calls `AuthorizeAddressExecute`.

Reply:
Ok, thats done, I left out `RevokeAddresses` because if someone already voted, we'd either have to fail, or give the owner god power to remove addresses and any votes that may have already be cast. 

Can add this method in, just have to decide what actions we want to take if the address being revoked has already voted.

Reply:
I guess the contract owner would already have god power by virtue of deciding who may vote.

Reply:
Further thoughts on this? @YakupIpek @quantumagi The more I think about it, the more I wonder if we really should add Revoke functionality at all.

Reply:
@mrtpain Either way it is fine. There are pros/cons of both. So i leave this decision for you.